sdk/src/transaction.rs

```diff
@@ -255,6 +255,61 @@ impl Transaction {
         }
     }
 
+    /// Check keys and keypair lengths, then sign this transaction, using an assortment of
+    /// KeypairUtil trait objects
+    pub fn sign_dynamic_signers(
+        &mut self,
```

Comment:
Why `mut`?

Reply:
This emulates the original `Transaction::sign` method to place the signature in the Transaction struct itself, hence self needs to be mut.
https://github.com/solana-labs/solana/pull/8336/files#diff-b8441031257e554aba72a1fd238539bcR308